normandy/recipes/api/serializers.py

```diff
@@ -21,10 +24,137 @@ class Meta:
         ]
 
 
+class ApprovalSerializer(serializers.ModelSerializer):
+    creator = UserSerializer(read_only=True)
+    creator_id = serializers.IntegerField(source='creator.id', write_only=True)
+
+    class Meta:
+        model = Approval
+        fields = [
+            'id',
+            'created',
+            'creator',
+            'creator_id',
+        ]
+
+    def validate_creator_id(self, attr):
+        try:
+            User.objects.get(id=attr)
+        except User.DoesNotExist:
+            raise serializers.ValidationError('User does not exist.')
+        return attr
+
+    def create(self, validated_data):
+        creator_id = validated_data.pop('creator')['id']
+        creator = User.objects.get(id=creator_id)
+        approval_request = Approval.objects.create(creator=creator, **validated_data)
+        return approval_request
+
+    def update(self, instance, validated_data):
+        instance.created = validated_data.get('created', instance.created)
+
+        if 'creator' in validated_data:
+            creator_id = validated_data.pop('creator')['id']
+            instance.action = User.objects.get(id=creator_id)
+
+        instance.save()
+        return instance
+
+
+class ApprovalRequestSerializer(serializers.ModelSerializer):
+    creator = UserSerializer(read_only=True)
+    creator_id = serializers.IntegerField(source='creator.id', write_only=True)
+    approval = ApprovalSerializer(required=False)
+
+    class Meta:
+        model = ApprovalRequest
+        fields = [
+            'id',
+            'created',
+            'creator',
+            'creator_id',
+            'active',
+            'approval',
+        ]
+
+    def validate_creator_id(self, attr):
+        try:
+            User.objects.get(id=attr)
+        except User.DoesNotExist:
+            raise serializers.ValidationError('User does not exist.')
+        return attr
+
+    def create(self, validated_data):
+        creator_id = validated_data.pop('creator')['id']
+        validated_data['creator'] = User.objects.get(id=creator_id)
+
+        if 'approval' in validated_data:
```

Comment:
Wouldn't an approval be created separately after an approval request? This might be unnecessary in that case.